dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -204,31 +210,79 @@ void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring
+
+        size_t lastBackslashPosition{ displayName.rfind(L"\\") };
+        THROW_HR_IF(E_UNEXPECTED, lastBackslashPosition == std::wstring::npos);
+        displayName = displayName.substr(lastBackslashPosition + 1); // One after the delimiter
+
+        displayName.erase(displayName.find_first_of(L".")); // Remove file extension
+    }
+
+    // Retrieve the icon
+    std::wstring iconFilePath{};
+
+    wil::unique_prop_variant propVariantIcon;
+    THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchIconResource, &propVariantIcon));
+    iconFilePath = propVariantIcon.pwszVal != nullptr ? propVariantIcon.pwszVal : L"";
+
+    // Icon filepaths from Shell APIs have this format: <filepath>,-<index>,
+    // since .ico files can have multiple icons in the same file.
+    // NotificationController doesn't seem to support such format, so let it take the first icon by default.
+    auto iterator = iconFilePath.find_first_of(L",");
+    if (iterator != std::wstring::npos)
+    {
+        iconFilePath.erase(iterator);
+    }
+
     RegisterValue(hKey, L"DisplayName", reinterpret_cast<const BYTE*>(displayName.c_str()), REG_EXPAND_SZ, displayName.size() * sizeof(wchar_t));
-    RegisterValue(hKey, L"IconUri", reinterpret_cast<const BYTE*>(iconUri.AbsoluteUri().c_str()), REG_EXPAND_SZ, iconUri.AbsoluteUri().size() * sizeof(wchar_t));
+    RegisterValue(hKey, L"IconUri", reinterpret_cast<const BYTE*>(iconFilePath.c_str()), REG_EXPAND_SZ, iconFilePath.size() * sizeof(wchar_t));
 
     std::wstring wideStringClsid{ clsid.get() };
     RegisterValue(hKey, L"CustomActivator", reinterpret_cast<const BYTE*>(wideStringClsid.c_str()), REG_SZ, wideStringClsid.size() * sizeof(wchar_t));
 }
 
-std::wstring Microsoft::Windows::AppNotifications::Helpers::RegisterComActivatorGuidAndAssets(winrt::Microsoft::Windows::AppNotifications::AppNotificationActivationInfo const& details)
+std::wstring Microsoft::Windows::AppNotifications::Helpers::RegisterComActivatorGuidAndAssets()
 {
     std::wstring registeredGuid;
     HRESULT status = GetActivatorGuid(registeredGuid);
     if (status == ERROR_FILE_NOT_FOUND)
     {
+        // Get process name to identify the app 
         wil::unique_cotaskmem_string processName;
         THROW_IF_FAILED(GetCurrentProcessPath(processName));
 
+        // Create a GUID for the COM Activator
         GUID comActivatorGuid = GUID_NULL;
         THROW_IF_FAILED(CoCreateGuid(&comActivatorGuid));
 
-        // StringFromCLSID returns GUID String with braces.
+        // StringFromCLSID returns GUID String with braces
         wil::unique_cotaskmem_string comActivatorGuidString;
         THROW_IF_FAILED(StringFromCLSID(comActivatorGuid, &comActivatorGuidString));
 
         std::wstring notificationAppId{ RetrieveNotificationAppId() };
-        RegisterAssets(notificationAppId, details.DisplayName(), details.IconUri(), comActivatorGuidString);
+        RegisterAssets(notificationAppId, comActivatorGuidString);
```

Comment:
This code has a serious bug. If the Display assets are set once, subsequent calls to Register()_ for unpackaged apps can never update the display assets again!